include/openthread.h

```diff
@@ -160,12 +174,117 @@ extern void otSignalTaskletPending(otInstance *aInstance);
  * @returns A pointer to the OpenThread version.
  *
  */
-const char *otGetVersionString(void);
+OTAPI const char *OTCALL otGetVersionString(void);
```

Comment:
I also would prefer to stay with current API, as an 3rd alternative if this change is really needed i can think of doing something like we do in BLE SoftDevice (kernel mode), where we communicate over SVCs.

`SVCALL(SD_BLE_GATTC_READ, uint32_t, sd_ble_gattc_read(uint16_t conn_handle, uint16_t handle, uint16_t offset));`

So it is modified version of your alternatives, that may give more flexibility. However what is in pull request is much simpler and probably better.